### ISSUE TITLE
Build Kubernetes on Windows host without make

### ISSUE THREAD

**Issue by mloskot** [kubernetes-sigs/sig-windows-dev-tools#255]
This is a TODO item from PR
- #254

as per @knabben suggestionin https://github.com/kubernetes-sigs/sig-windows-dev-tools/pull/254#discussion_r1173892296

Basically, this procedure 

https://github.com/kubernetes-sigs/sig-windows-dev-tools/blob/171f2ca763ea90d19e996bcc28a055ba1086c09b/build.sh#L45-L52

needs to be ported as `Invoke-Build` in `make.ps1` but as Windows-native procedure building Kubernetes from source without using `make`, that is without WSL, but on Windows host directly.

**Comment by claudiubelu** [comment]
I wonder if we can simply do ``go build -o binary.exe .\cmd\binary`` on Windows. We're already building kubeadm.exe on Windows for unit tests (https://github.com/kubernetes-sigs/windows-testing/blob/master/scripts/k8s_unit_windows.ps1#L83), although, to be fair, some flags may need to be set as well.

**Comment by aravindhp** [comment]
Given we are hoping to use magefiles on Linux also, will using `go build` result in a static or dynamically linked executable? Because with Linux we could be building on distro X and running the executable on Ubuntu which will not work if it dynamically linked.

**Comment by mloskot** [comment]
@aravindhp 
> Given we are hoping to use magefiles on Linux also, will using go build result in a static or dynamically linked executable?

I wrote my first line in Go a few days ago, so I don't know what are build and cross-build options in Go.
So, build details need to be worked out.

But, here is my plan in bigger picture:

```
$ mage -l | grep -E 'build|fetch'
  build              Clone and build Kubernetes from source (on Windows native host use fetch).
  fetch              Download Kubernetes binaries.
```

which means this:
- on Linux, Mac OS and WSL too, users will be able to download binaries or build from source
- on Windows native (no WSL) users will only be able to download binaries

**Comment by jsturtevant** [comment]
We should be able to use `go build -o binary.exe .\cmd\binary` on Windows as suggested by @claudiubelu for the windows scenario so they can build them as well.  I've done this in the past. Is using the make files for linux and go build for Windows feasible?  That is the approach I would suggest. Otherwise you could do something like running the command on wsl and coping it locally https://stackoverflow.com/a/66154000/697126.

**Comment by mloskot** [comment]
@jsturtevant 
> Is using the make files for linux and go build for Windows feasible? 

Yes, I think that is feasible. I will try to implement it and let's see how it works.

**Comment by aravindhp** [comment]
@sbdtu5498 please see https://github.com/kubernetes-sigs/sig-windows-dev-tools/tree/master-windows-native, where there work in progress to make mage files works. Contributions are welcome.

**Comment by mloskot** [comment]
Closing this as being replaced by the currently ongoing SWDT CLI sub-project
1. #278 (see #277 for context)
2. https://opssec.in/2024/0104/ by @knabben on cross platform build part
